src/Microsoft.ML.Data/Training/TrainerUtils.cs

```diff
@@ -348,6 +348,41 @@ public static ValueGetter<ulong> GetOptGroupGetter(this IRow row, RoleMappedData
             Contracts.CheckValue(data, nameof(data));
             return GetOptGroupGetter(row, data.Schema);
         }
+
+        public static SchemaShape.Column MakeBoolScalarLabel(string labelColumn)
+        {
+            return new SchemaShape.Column(labelColumn, SchemaShape.Column.VectorKind.Scalar, BoolType.Instance, false);
+        }
+
+        public static SchemaShape.Column MakeR4ScalarLabel(string labelColumn)
+        {
+            return new SchemaShape.Column(labelColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false);
+        }
+
+        public static SchemaShape.Column MakeU4ScalarLabel(string labelColumn)
+        {
+            return new SchemaShape.Column(labelColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.U4, true);
+        }
+
+        /// <summary>
+        /// The <see cref="SchemaShape.Column"/> for the feature column.
+        /// </summary>
+        /// <param name="featureColumn">name of the feature column</param>
+        public static SchemaShape.Column MakeR4VecFeature(string featureColumn)
+        {
+            return new SchemaShape.Column(featureColumn, SchemaShape.Column.VectorKind.Vector, NumberType.R4, false);
+        }
+
+        /// <summary>
+        /// The <see cref="SchemaShape.Column"/> for the feature column.
+        /// </summary>
+        /// <param name="weightColumn">name of the feature column</param>
```

Comment:
>name of the feature column [](start = 39, length = 26)

Description does not agree to name of the parameter...:)

Reply:
Also, I see that some of the methods have comments while others don't.

---
In reply to: [219253274](https://github.com/dotnet/machinelearning/pull/962#discussion_r219253274) [](ancestors = 219253274)